x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SslSettingsLoader.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ssl;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.SecureString;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.ssl.SslClientAuthenticationMode;
+import org.elasticsearch.common.ssl.SslConfigException;
+import org.elasticsearch.common.ssl.SslConfiguration;
+import org.elasticsearch.common.ssl.SslConfigurationLoader;
+import org.elasticsearch.common.ssl.SslKeyConfig;
+import org.elasticsearch.common.ssl.SslTrustConfig;
+import org.elasticsearch.common.ssl.SslVerificationMode;
+import org.elasticsearch.env.Environment;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * A configuration loader for SSL Settings
+ */
+public class SslSettingsLoader extends SslConfigurationLoader {
+
+    private final Settings settings;
+    private final Map<String, Setting<? extends SecureString>> secureSettings;
+    private final Map<String, Setting<?>> standardSettings;
+    private final Map<String, Setting<?>> ignoredSettings;
+
+    public SslSettingsLoader(Settings settings, String settingPrefix, boolean acceptNonSecurePasswords) {
+        super(settingPrefix);
+        this.settings = settings;
+        final SSLConfigurationSettings sslConfigurationSettings = settingPrefix == null
+            ? SSLConfigurationSettings.withoutPrefix(acceptNonSecurePasswords)
+            : SSLConfigurationSettings.withPrefix(settingPrefix, acceptNonSecurePasswords);
+        this.secureSettings = mapOf(sslConfigurationSettings.getSecureSettings());
+        this.standardSettings = mapOf(sslConfigurationSettings.getEnabledSettings());
+        this.ignoredSettings = mapOf(sslConfigurationSettings.getDisabledSettings());
```

Comment:
nit: disabledSettings so that the reader doesn't need to check what is disabled, what is ignored and if there is a difference?